liquid-staking/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+#![no_std]
+
+use multiversx_sc::imports::*;
+
+pub mod common;
+pub mod delegation;
+pub mod delegation_proxy;
+pub mod liquid_staking_proxy;
+pub mod liquidity_pools;
+
+#[multiversx_sc::contract]
+pub trait LiquidStaking:
+    liquidity_pools::LiquidityPoolModule
+    + delegation::DelegationModule
+    + common::storage::CommonStorageModule
+{
+    #[init]
+    fn init(&self) {}
+
+    #[endpoint(registerDelegationContractAddress)]
+    fn register_delegation_address(
+        &self,
+        contract_name: ManagedBuffer,
+        delegation_address: ManagedAddress,
+    ) {
+        require!(
+            self.blockchain().is_smart_contract(&delegation_address),
+            "Provided address is not a valid Smart Contract address"
+        );
+
+        require!(
+            self.delegation_addresses(&contract_name).is_empty(),
+            "This contract is already registered"
+        );
+
+        self.delegation_addresses(&contract_name)
+            .set(delegation_address);
+    }
+
+    #[only_owner]
+    #[endpoint(registerHeaderVerifierAddress)]
+    fn register_header_verifier_address(&self, header_verifier_address: ManagedAddress) {
+        self.header_verifier_address().set(header_verifier_address);
+    }
```

Comment:
I think this could be simply on init function instead of requiring one additional call. You can keep the function in case you ever need to overwrite the address.

Reply:
I wanted to keep the init empty at the moment since there will probably be a lot more to put inside that